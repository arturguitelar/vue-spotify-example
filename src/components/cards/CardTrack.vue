<template>
  <div class="card">
    <h3>{{ track.name }}</h3>
    <img :src="track.imageUrl" />
    <div class="info">
      <p>Artista: {{ track.artist }}</p>
    </div>
  </div>
</template>

<script>
export default {
  name: "CardTrack",
  props: {
    track: Object
  }
};
</script>
