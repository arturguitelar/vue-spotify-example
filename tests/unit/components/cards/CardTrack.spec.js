import { shallowMount } from "@vue/test-utils";
import CardTrack from "@/components/cards/CardTrack";

const DATASET_CARD = "[data-card='track']";
const DATASET_MESSAGE = "[data-card-message='empty']";
const IMAGE_URL = "https://via.placeholder.com/300";
const INVALID_DATA_MESSAGE = "Dados invÃ¡lidos";

describe("#CardTrack", () => {
  describe("#FAIL", () => {
    test("render invalid data if doesn't receive correct object", () => {
      const wrapper = shallowMount(CardTrack);

      expect(wrapper.find(DATASET_CARD).exists()).toBe(false);
      expect(wrapper.find(DATASET_MESSAGE).exists()).toBe(true);
      expect(wrapper.text()).toBe(INVALID_DATA_MESSAGE);
    });
  });

  describe("#SUCCESS", () => {
    const defaultData = {
      name: "Track name",
      imageUrl: IMAGE_URL,
      artist: "Artist name"
    };

    let wrapper;

    beforeEach(() => {
      wrapper = shallowMount(CardTrack, {
        props: { track: defaultData }
      });
    });

    afterEach(() => {
      wrapper.unmount();
    });

    test("create a card with correct data", () => {
      expect(wrapper.vm.track).toStrictEqual(defaultData);
      expect(wrapper.props()).toEqual({ track: defaultData });
      expect(wrapper.find(DATASET_CARD).exists()).toBe(true);
      expect(wrapper.find(DATASET_MESSAGE).exists()).toBe(false);
    });

    test("image should be placed by placeholder if no image url", () => {
      defaultData.imageUrl = "";

      wrapper = shallowMount(CardTrack, {
        props: { track: defaultData }
      });

      expect(wrapper.find("img").attributes()).toEqual({
        src: IMAGE_URL
      });
    });

    describe("Snapshots", () => {
      test("should render a card correctly", () => {
        expect(wrapper.wrapperElement).toMatchSnapshot();
      });
    });
  });
});
