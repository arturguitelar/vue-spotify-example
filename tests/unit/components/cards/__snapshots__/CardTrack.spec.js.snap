// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#CardTrack #SUCCESS Snapshots should render a card correctly 1`] = `
<div
  class="card card-track"
  data-card="track"
>
  <img
    src="https://via.placeholder.com/300"
  />
  <div
    class="info"
  >
    <h3>
      Track name
    </h3>
    <p
      class="artist"
    >
      Artist name
    </p>
  </div>
</div>
`;
